@stuartwdouglas @dmlloyd I can see some native failures here and I have no idea what to do about that. Could you please help?

The stack from CI goes like this:
> Caused by: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: java.lang.Object.toString(). To diagnose the issue you can use the --allow-incomplete-classpath option. The missing method is then reported at run time when it is accessed the first time.

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ClientProxyGenerator.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ClientProxyGenerator.java
@@ -134,7 +134,10 @@ public class ClientProxyGenerator extends AbstractGenerator {
              * as it just works, and the reflection case cannot be true since it's not possible to have
              * non-public default interface methods.
              */
-            if (isInterface) {
+            if (Methods.isObjectToString(method)) {
+                // Always use invokevirtual and the original descriptor for java.lang.Object#toString()
+                ret = forward.invokeVirtualMethod(originalMethodDescriptor, delegate, params);
+            } else if (isInterface) {
                 ret = forward.invokeInterfaceMethod(method, delegate, params);
             } else if (isReflectionFallbackNeeded(method, targetPackage)) {
                 // Reflection fallback

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java
@@ -107,6 +107,10 @@ final class Methods {
         return false;
     }
 
+    static boolean isObjectToString(MethodInfo method) {
+        return method.declaringClass().name().equals(DotNames.OBJECT) && method.name().equals(TO_STRING);
+    }
+
     static void addInterceptedMethodCandidates(BeanDeployment beanDeployment, ClassInfo classInfo,
             Map<MethodKey, Set<AnnotationInstance>> candidates,
             List<AnnotationInstance> classLevelBindings) {

